pkg/backends/azurekeyvault.go

```diff
@@ -0,0 +1,99 @@
+package backends
+
+import (
+	"context"
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/keyvault/keyvault"
+	"path"
+	"strings"
+	"time"
+)
+
+// AzureKeyVault is a struct for working with an Azure Key Vault backend
+type AzureKeyVault struct {
+	Client keyvault.BaseClient
+}
+
+// NewAzureKeyVaultBackend initializes a new Azure Key Vault backend
+func NewAzureKeyVaultBackend(client keyvault.BaseClient) *AzureKeyVault {
+	return &AzureKeyVault{
+		Client: client,
+	}
+}
+
+// Login does nothing as a "login" is handled on the instantiation of the Azure SDK
+func (a *AzureKeyVault) Login() error {
+	return nil
+}
+
+// GetSecrets gets secrets from Azure Key Vault and returns the formatted data
+func (a *AzureKeyVault) GetSecrets(kvpath string, version string, _ map[string]string) (map[string]interface{}, error) {
+	kvpath = "https://" + kvpath + ".vault.azure.net"
```

Comment:
Not a big deal, but should `fmt.Sprintf` be used here instead?

Reply:
Agree 👍 fixing